starlette/requests.py

```diff
@@ -187,3 +190,15 @@ def receive(self) -> Receive:
             for item in self._form.values():
                 if hasattr(item, "close"):
                     await item.close()  # type: ignore
+
+    async def is_disconnected(self) -> bool:
+        if not self._is_disconnected:
+            try:
+                message = await asyncio.wait_for(self._receive(), timeout=0.0000001)
```

Comment:
In https://github.com/encode/starlette/issues/297#issuecomment-454024709 `timeout=0` was used.
What's the difference?

And what about using just self._receive.disconnected?  (https://github.com/encode/starlette/issues/297#issuecomment-451089517)

Reply:
> In #297 (comment) timeout=0 was used. What's the difference?

Turns out that doesn't actually work (according to the tests). Needs to be non-zero.

> And what about using just self._receive.disconnected?

That's an implementation detail that'll only happen to work with uvicorn, in current versions. The API contract between the ASGI server and the client is just "recieve is an awaitable that returns messages".